src/spyglass/spikesorting/v1/recording.py

```diff
@@ -202,43 +208,160 @@ def make(self, key):
+
+        Parameters
+        ----------
+        key : dict
+            primary key of SpikeSortingRecordingSelection table
+        recompute_file_name : str, Optional
+            If specified, recompute this file. Use as resulting file name.
+            If none, generate a new file name. Used for recomputation after
+            typical deletion.
+        save_to : Union[str,Path], Optional
+            Default None, save to analysis directory. If provided, save to
+            specified path. Used for recomputation prior to deletion.
+        """
+        if not key and not recompute_file_name:
+            raise ValueError(
+                "Either key or recompute_file_name must be specified."
+            )
+        if isinstance(key, dict):
+            key = {k: v for k, v in key.items() if k in cls.primary_key}
+
+        hash = None
+        recompute = recompute_file_name and not key and not save_to
+
+        if recompute or save_to:  # if we expect file to exist
+            file_path = AnalysisNwbfile.get_abs_path(
+                recompute_file_name, from_schema=True
+            )
+        if recompute:  # If recompute, check if file exists
+            if Path(file_path).exists():  # No need to recompute
+                return
+            logger.info(f"Recomputing {recompute_file_name}.")
+            query = cls & {"analysis_file_name": recompute_file_name}
+            # Use deleted file's ids and hash for recompute
+            key, recompute_object_id, recompute_electrodes_id, hash = (
+                query.fetch1("KEY", "object_id", "electrodes_id", "hash")
+            )
+        elif save_to:  # recompute prior to deletion, save copy to temp_dir
+            elect_id = cls._validate_file(file_path)
+            obj_id = (cls & key).fetch1("object_id")
+            recompute_object_id, recompute_electrodes_id = obj_id, elect_id
+        else:
+            recompute_object_id, recompute_electrodes_id = None, None
+
+        parent = SpikeSortingRecordingSelection & key
+        (recording_nwb_file_name, recording_object_id, electrodes_id) = (
+            _write_recording_to_nwb(
+                **cls()._get_preprocessed_recording(key),
+                nwb_file_name=parent.fetch1("nwb_file_name"),
+                recompute_file_name=recompute_file_name,
+                recompute_object_id=recompute_object_id,
+                recompute_electrodes_id=recompute_electrodes_id,
+                save_to=save_to,
+            )
         )
-        AnalysisNwbfile().log(
-            recording_nwb_file_name, table=self.full_table_name
+
+        hash = AnalysisNwbfile().get_hash(
+            recording_nwb_file_name,
+            from_schema=True,
+            precision_lookup=rounding,
+            return_hasher=bool(save_to),
+        )
+
+        # NOTE: Conditional to avoid impacting database. NO MERGE!
+        if recompute and test_mode:
+            AnalysisNwbfile()._update_external(recompute_file_name, hash)
```

Comment:
Can you explain why this is here?

Reply:
This and the current schema names are intended to preserve the database state while under review. This conditional prevents a tester from accidentally updating the external table while we're still reviewing. Once things are by-and-large approved, I'll remove this conditional and rename the schemas